### PR TITLE
DDF-1532:Makes Subject DN Constraints Configurable

### PR DESCRIPTION
**Pull request by vinamartin**
-Added a system property that allow admins to set the STS Subject DN Constraints. By default it is ".*".
-Some of the security itests now use authorization

Reviewers: 
@stustison @djblue @clockard @ricklarsen @gordocanchola

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/267)

### CODE REVIEW

File: distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java [hunk 1]
```diff
@@ -170,6 +185,12 @@ public void beforeTest() throws Exception {
         waitForAllBundles();
         configurePDP();
         waitForHttpEndpoint(SERVICE_ROOT + "/catalog/query");
+        keystore = KeyStore.getInstance("jks");
```
Comment (clockard) [inline distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java]: What is the keystore getting used for? I don't see it being used anywhere. Same with the truststore

File: platform/security/sts/security-sts-server/src/main/java/ddf/security/sts/SubjectDNConstraintsInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import org.apache.cxf.message.Message;
+import org.apache.cxf.phase.AbstractPhaseInterceptor;
+import org.apache.cxf.phase.Phase;
+import org.apache.cxf.ws.security.wss4j.AbstractWSS4JInterceptor;
+import org.apache.wss4j.dom.handler.WSHandlerConstants;
+import org.eclipse.jetty.server.Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SubjectDNConstraintsInterceptor extends AbstractPhaseInterceptor<Message> {
```
Comment (clockard) [inline platform/security/sts/security-sts-server/src/main/java/ddf/security/sts/SubjectDNConstraintsInterceptor.java]: A little bit of javadoc at the class level might be nice to explain what this interceptor is doing here.

File: distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java (hunk 1 above)
Comment (vinamartin) [inline distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java]: I think I originally made them to do the auth (like on line 389) but now that I look at it, it looks like they only need the paths and the password. I'll remove them.

File: distribution/test/itests/test-itests-catalog/pom.xml [hunk 3]
```diff
@@ -70,6 +70,18 @@
             <scope>test</scope>
         </dependency>
         <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>com.springsource.org.apache.httpcomponents.httpcore</artifactId>
+            <version>4.2.1</version>
```
Comment (coyotesqrl) [inline distribution/test/itests/test-itests-catalog/pom.xml]: ddf-platform's pom uses version 4.3.3 of `httpcore` and 4.3.5 of `httpclient`. Is there a reason you're specifying older versions here?

File: distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java [hunk 4]
```diff
@@ -223,6 +223,18 @@ protected UrlResourceReaderConfigurator getUrlResourceReaderConfigurator() {
         return options(junitBundles(),
                 // HACK: incorrect version exported to override hamcrest-core from exam
```
Comment (coyotesqrl) [inline distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java]: This comment should be moved down so it is connected to the line it explains.

File: distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java [hunk 5]
```diff
@@ -223,6 +223,18 @@ protected UrlResourceReaderConfigurator getUrlResourceReaderConfigurator() {
...
                 // feature which causes a split package issue for rest-assured
+                wrappedBundle(mavenBundle("org.apache.httpcomponents",
+                        "com.springsource.org.apache.httpcomponents.httpclient")
```
Comment (coyotesqrl) [inline distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java]: Maybe my earlier comment in the pom file was misunderstanding on my part. Are the `com.springsource.org.apache.httpcomponents.httpclient` and `httpclient` artifacts not the same but with different names on the newer versions? (Same question re: the two `httpcore` artifacts.)

Comment (vinamartin) [inline distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java]: Yes, the springsource versions have already been made into bundles.

Comment (coyotesqrl) [inline distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java]: Thanks for the clarification.

File: platform/security/sts/security-sts-server/src/main/java/ddf/security/sts/SubjectDNConstraintsInterceptor.java (hunk 2 above)
Comment (vinamartin) [inline platform/security/sts/security-sts-server/src/main/java/ddf/security/sts/SubjectDNConstraintsInterceptor.java]: Updated. Let me know if I should add more or what you think would be more helpful.

**Comment by coyotesqrl** [comment]
LGTM.

**Comment by clockard** [comment]
:+1:

**Comment by gordocanchola** [comment]
:+1:

**Comment by roelens8** [comment]
Built successfully, all tests pass! :+1:

**Comment by roelens8** [comment]
Built successfully again and all tests passed!

**Pull request merged**